### PR TITLE
Staging -> Production

### PR DESCRIPTION
**Pull request by maxharrisnet**
**Changes**

- Adding CMS to all pages with `pages.js`
- Pulling Team Members from the CMS in `team.js`
-- Constructing cards and modals for team

** Reviewing Code**
The JavaScript is the main code that needs to be reviewed (class and id changes are just to standardize components and keep JS working). I will add comments for clarification.

**Testing CMS**
- Staging: https://staging.iftheselandscouldtalk.org/ (current code for testing)
- Production: https://iftheselandscouldtalk.org/ (main)
- WordPress CMS Admin: https://wp.iftheselandscouldtalk.org/wp-admin
- Contact Max Harris for WP account setup
- Edit content in WordPress, view changes on corresponding pages/sections in staging. All page content should be editable.
- Admin > Pages to pages
- Admin > Events (this is currently working in production, so be careful with publishing)

Screen Shot 2024-02-02 at 11 45 17 AM
Screen Shot 2024-02-02 at 11 45 26 AM
Screen Shot 2024-02-02 at 11 46 44 AM

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -7,64 +7,6 @@ Modals
 // HTML Reference Variables
 let openModalBtns = document.querySelectorAll(`[data-role="open-modal"]`);
 let closeModalBtns = document.querySelectorAll(`[data-role="close-modal"]`);
-
```
Comment (maxharrisnet) [inline js/main.js:10]: Moving all of this to `team.js`

File: js/main.js [hunk 2]
```diff
@@ -76,20 +18,8 @@ const toggleModal = modal => {
 openModalBtns.forEach(button => {
 	button.addEventListener("click", () => {
 		let modalElement = document.querySelector(`#${button.dataset.type}`);
-
-		let member = button.dataset.content;
```
Comment (maxharrisnet) [inline js/main.js:80]: Moving the member stuff to `team.js`

File: js/main.js [hunk 3]
```diff
@@ -148,3 +78,65 @@ mobileMenuBtn.addEventListener("click", () => {
 	// toggle mobile menu component
 	mobileMenuComponent.classList.toggle("open-menu");
 });
+
+/* 
+===================================
+Content Management System
+===================================
+*/
+// Helper functions for appending data to DOM elements
```
Comment (maxharrisnet) [inline js/main.js:87]: This is meant to keep all the appending as DRY as possible and to run checks. The function works but could probably be improved (moving quickly).

File: js/pages.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	fetch(`${apiURL}`, {
+		method: "GET",
+		headers: {
+			"Content-Type": "application/json",
+		},
+	})
+		.then(response => response.json())
+		.then(response => {
+			// Page data
+			const data = response[0].acf;
```
Comment (maxharrisnet) [inline js/pages.js:14]: Advanced Custom Fields is the WordPress plugin that is used to create the fields. All of the data we need is coming is from that object (except the title).

File: js/pages.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
+// Fetch page data from WordPress
```
Comment (maxharrisnet) [inline js/pages.js:1]: This file is for all pages. It fetches the data for that page. If new data exists from the CMS, we append it to some element with `appendData()` in `main.js`.

Using IDs to target most things.
Data from the CMS is set  up to use the same fields on each page (i.e. `section_1`, `hero_image`) and the pages use the same IDs (i.e. `#section-1`, `#hero-image`).

File: js/pages.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+				? appendData(data.hero_subheading, "h2", "text", heroSubheading)
+				: null;
+			data.hero_button_1
+				? appendData(data.hero_button_1, "hero-button-1", "link", heroButton1)
+				: null;
+			data.hero_button_1
+				? appendData(data.hero_button_2, "hero-button-2", "link", heroButton2)
+				: null;
+
+			// Main Content Sections
```
Comment (maxharrisnet) [inline js/pages.js:44]: This is used on about half of the pages - those with a big single area for content like About Us. 

Screen Shot 2024-02-02 at 12 20 02 PM

File: js/pages.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+			const mainContent = document.getElementById("main-content");
+			data.main_content
+				? appendData(data.main_content, "div", "markup", mainContent)
+				: null;
+
+			// Generic Content Sections
```
Comment (maxharrisnet) [inline js/pages.js:50]: These are the mostly 2-column sections on many pages (Homepage, Impact, etc.)

File: js/pages.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+						: null;
+					sectionData.text && sectionText
+						? appendData(sectionData.text, "p", "text", sectionText)
+						: null;
+					sectionData.quote && sectionQuote
+						? appendData(sectionData.quote, "blockquote", "text", sectionQuote)
+						: null;
+
+					if (sectionData.button && sectionButton) {
+						// appendData(sectionData.button, "button", "link", sectionButton);
```
Comment (maxharrisnet) [inline js/pages.js:84]: Not working, appends the wrong way. No time to fix. Must complete project. Must warn others...

File: js/team.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
+// Populate modal
```
Comment (maxharrisnet) [inline js/team.js:1]: This file fetches all of the team members from the CMS, generates card components for each, and generates modal data for each.

File: js/team.js [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+function populateTeamModal(data) {
```
Comment (maxharrisnet) [inline js/team.js:2]: Making the modal data dynamic

File: js/team.js [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+	// text
+	memberName.textContent = name;
+	memberRole.textContent = role;
+
+	// bio
+	appendData(bio, "p", "text", memberBio);
+}
+
+function generateCards(items) {
```
Comment (maxharrisnet) [inline js/team.js:30]: Creating card components to append to the container. 

Screen Shot 2024-02-02 at 12 21 28 PM

File: js/team.js [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+		if (data.bio && text) {
+			const btn = document.createElement("button");
+			btn.classList.add("btn-main", "team-modal-btn");
+			btn.setAttribute("data-role", "open-team-modal");
+			btn.setAttribute("data-type", "team-modal");
+			btn.setAttribute("data-content", slugify(data.name));
+			btn.textContent = "Read More";
+			text.appendChild(btn);
+
+			btn.addEventListener("click", () => {
```
Comment (maxharrisnet) [inline js/team.js:79]: Event listener for populating and opening modals. 

Screen Shot 2024-02-02 at 12 22 12 PM

File: js/team.js [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+			});
+		}
+
+		// Append this card to the container
+		container.appendChild(card);
+	});
+}
+
+// Fetch all team members
+function getAllTeamData() {
```
Comment (maxharrisnet) [inline js/team.js:104]: 🐶

File: team.html [hunk 14]
```diff
@@ -112,71 +113,12 @@ <h1 id="hero-heading">Our Team</h1>
 		<!-- Team Members -->
 		<section>
 			<div class="wrapper">
-				<ul class="team-members">
```
Comment (maxharrisnet) [inline team.html:115]: Replacing static content with dynamic

File: js/cms.js
Comment (maxharrisnet) [inline js/cms.js:1]: This was broken up into the other JS files

File: js/pages.js [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+						sectionButton.href = sectionData.button.url;
+						sectionButton.textContent = sectionData.button.title;
+					}
+				} else {
+					null;
+				}
+			});
+
+			// Cards
```
Comment (maxharrisnet) [inline js/pages.js:93]: 2 and 3 column card elements (Values page)
Screen Shot 2024-02-02 at 12 17 56 PM

File: js/pages.js [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+		method: "GET",
+		headers: {
+			"Content-Type": "application/json",
+		},
+	})
+		.then(response => response.json())
+		.then(response => {
+			// Page data
+			const data = response[0].acf;
+			// Hero
```
Comment (maxharrisnet) [inline js/pages.js:15]: Hero

Screen Shot 2024-02-02 at 12 18 30 PM

File: js/pages.js (hunk 6 above)
Comment (maxharrisnet) [reply js/pages.js:44]: Screen Shot 2024-02-02 at 12 19 15 PM

**Review by isaackIFA — CHANGES_REQUESTED** [review]
Looks good! WordPress CMS creation, editing and deleting are all working for pages, events and team members. All changes show up on website without any issues.

There are a 2 small UI issues on the "Our Team" page
- Duplicate sentence under "Our Current Positions" SectionScreenshot 2024-02-06 at 10 50 53 AM

- Homepage button links to default/blank wordpress pageScreenshot 2024-02-06 at 11 08 17 AM

**Pull request merged**